Doc/library/sqlite3.rst

```diff
@@ -47,84 +47,152 @@ This document includes four main sections:
+including `cursors`_ and `transactions`_.
+
+First, create a new database to hold the movie data,
+and open a database connection to allow :mod:`!sqlite3` to work with it.
+Call the :func:`sqlite3.connect` function to
+to create a connection to the database :file:`tutorial.db`,
+implicitly creating it in the current working directory if it does not exist::
 
    import sqlite3
-   con = sqlite3.connect('example.db')
+   con = sqlite3.connect("tutorial.db")
 
-The special path name ``:memory:`` can be provided to create a temporary
-database in RAM.
+The returned :class:`Connection` object ``con``
+represents the connection to the on-disk database.
 
-Once a :class:`Connection` has been established, create a :class:`Cursor` object
-and call its :meth:`~Cursor.execute` method to perform SQL commands::
+In order to execute SQL statements and fetch results from SQL queries,
+use a database cursor.
+Call :meth:`con.cursor() <Connection.cursor>` to create the :class:`Cursor`::
 
    cur = con.cursor()
 
-   # Create table
-   cur.execute('''CREATE TABLE stocks
-                  (date text, trans text, symbol text, qty real, price real)''')
+Now, create a database table ``movie`` with columns for title,
+release year, and review score.
+The `flexible typing`_ feature of SQLite makes typenames optional;
+for simplicity, just use column names in the table declaration.
+Execute the SQL statement
+by calling :meth:`con.execute() <Cursor.execute>`::
 
-   # Insert a row of data
-   cur.execute("INSERT INTO stocks VALUES ('2006-01-05','BUY','RHAT',100,35.14)")
+   cur.execute("CREATE TABLE Movie(Title, Year, Score)")
 
-   # Save (commit) the changes
-   con.commit()
+.. Ideally, we'd use sqlite_schema instead of sqlite_master below,
+   but earlier versions of SQLite does not recognise that variant.
 
-   # We can also close the connection if we are done with it.
-   # Just be sure any changes have been committed or they will be lost.
-   con.close()
+You can verify that the table has been created by querying
+the ever-present ``sqlite_master`` table.
+It should contain an entry for the ``Movie`` table definition.
+Execute the query by calling :meth:`cur.execute(...) <Cursor.execute>`,
+store the result in a variable ``res``,
+and call :meth:`res.fetchone() <Cursor.fetchone>` to fetch the first
+(and only) row from the query::
 
-The saved data is persistent: it can be reloaded in a subsequent session even
-after restarting the Python interpreter::
+   >>> res = cur.execute("SELECT name FROM sqlite_master where name='Movie'")
+   >>> res.fetchone()
+   ('Movie',)
 
-   import sqlite3
-   con = sqlite3.connect('example.db')
-   cur = con.cursor()
+The result is a one-item :class:`tuple`: one row, with one column.
+As expected, the query shows the table is now created.
+As an exercise, try querying ``sqlite_master`` for a bogus name ``"abc"``::
+
+   >>> res = cur.execute("SELECT name FROM sqlite_master where name='abc'")
```

Comment:
```suggestion
   >>> res = cur.execute("SELECT name FROM sqlite_master where name='movie'")
```
Is `sqlite3` case-sensitive?  If it is, instead of using bogus tables, you might use this and get two birds with a stone (teach about case-sensitiveness and show what happens when a query returns no results).

Reply:
Most SQL dialects are case-insensitive in most ways. SQLite has some pragma options for case-sentitivity. Perhaps there are also some compile-time options.